ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/TransactionExtension.java

```diff
@@ -91,20 +89,14 @@ public void processManagedBean(@Observes ProcessManagedBean<?> pmb) {
         beanToAnnotatedTypeMapping.put(pmb.getBean(), pmb.getAnnotatedBeanClass());
     }
 
-    public void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager) {        
-
-        Set<Bean<?>> beans = manager.getBeans(TransactionSynchronizationRegistry.class);
-        if (beans == null || beans.isEmpty()) {
-            event.addBean(new JNDIBean<>(jtaPropertyManager.getJTAEnvironmentBean().getTransactionSynchronizationRegistryJNDIContext(), TransactionSynchronizationRegistry.class, Singleton.class));
-        }
-        
+    public void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager) {
         event.addContext(new TransactionContext(() -> {
             final Bean<?> tmBean = manager.resolve(manager.getBeans(TransactionManager.class));
-            return (TransactionManager)manager.getReference(tmBean, TransactionManager.class, manager.createCreationalContext(tmBean));
+            return (TransactionManager) manager.getReference(tmBean, TransactionManager.class, manager.createCreationalContext(tmBean));
         },
         () -> {
             final Bean<?> tsrBean = manager.resolve(manager.getBeans(TransactionSynchronizationRegistry.class));
-            return (TransactionSynchronizationRegistry)manager.getReference(tsrBean, TransactionSynchronizationRegistry.class, manager.createCreationalContext(tsrBean));
+            return (TransactionSynchronizationRegistry) manager.getReference(tsrBean, TransactionSynchronizationRegistry.class, manager.createCreationalContext(tsrBean));
```

Comment:
Why aren't we also producing UserTransaction?

Reply:
A CDI implementation is [already required to produce `UserTransaction`](http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#additional_builtin_beans).  Weld does this through its [`TransactionServices` SPI interface](https://static.javadoc.io/org.jboss.weld/weld-spi/3.1.Final/org/jboss/weld/transaction/spi/TransactionServices.html#getUserTransaction--).